remove deer czm and update related routines

arovinelli opened the pull request:
Upgrade deer to use the new cohesive zone model available in MOOSE.

This requires a lot of changes to materials and to a variety of related routines

reverendbedford commented on src/materials/czm/CZMVolumetricStrain.C:
```diff
@@ -9,176 +9,86 @@ InputParameters CZMVolumetricStrain::validParams() {
+      _czm_sliding_strain(declarePropertyByName<RankTwoTensor>(
+          _base_name + "czm_sliding_strain")),
+      _dadA_mp(declarePropertyByName<Real>(_base_name + "czm_area_ratio")),
+      // _czm_total_strain_old(getMaterialPropertyOldByName<RankTwoTensor>(
```
Why are these commented out?

arovinelli replied:
we don't need these extra material properties, rates will be computed using post processor, faster and less memory usage.

reverendbedford commented on src/materials/czm/CZMVolumetricStrain.C:
```diff
@@ -9,176 +9,86 @@ InputParameters CZMVolumetricStrain::validParams() {
+                            _dadA_mp[_qp];
+  _czm_sliding_strain[_qp] = _czm_total_strain[_qp] - _czm_normal_strain[_qp];
+
+  // _czm_total_strain_rate[_qp] =
```
Why commented out?

arovinelli replied:
see above

reverendbedford commented on src/postprocessors/TimeDerivativePostprocessor.C:
```diff
@@ -0,0 +1,37 @@
+InputParameters TimeDerivativePostprocessor::validParams() {
+  InputParameters params = GeneralPostprocessor::validParams();
+  params.addClassDescription(
+      "Computest the time deribative of a postprocessor");
```
Compute

arovinelli replied:
done

reverendbedford reviewed (changes requested):
1. The homogenization tests failing in debug mode is my fault.  I already fixed that over in mainline tensor_mechanics, but need to fix it here.  I'll do that after you take care of these changes.
2. Can you add:
a. A few tests covering large versus small deformations (or modify one of your existing tests)
b. Add an example adding CZM to with the action (can be simple, just pull blocks or something)
3. A few minor comments in the review

arovinelli commented on include/postprocessors/MultiplyPostprocessor.h:
```diff
@@ -0,0 +1,36 @@
+template <> InputParameters validParams<MultiplyPostprocessor>();
+
+/**
+ * Computes the difference between two postprocessors
```
fixed

arovinelli commented on include/postprocessors/TimeDerivativePostprocessor.h:
```diff
@@ -0,0 +1,43 @@
+template <> InputParameters validParams<TimeDerivativePostprocessor>();
+
+/**
+ * Creates a cumulative sum of a post-processor value over a transient.
```
fixed

arovinelli commented on src/actions/CZMStrainAction.C:
```diff
@@ -23,34 +23,28 @@ InputParameters CZMStrainAction::validParams() {
   params.addParam<PostprocessorName>(
```
well numbers make sense

arovinelli commented:
> * The homogenization tests failing in debug mode is my fault.  I already fixed that over in mainline tensor_mechanics, but need to fix it here.  I'll do that after you take care of these changes.
> * Can you add:
>   a. A few tests covering large versus small deformations (or modify one of your existing tests)
>   b. Add an example adding CZM to with the action (can be simple, just pull blocks or something)
> * A few minor comments in the review

1. I didn't add CZMAction tests as they are all in the MOOSE mainline
2. added tests for CZM materials using large deformations
3. I resolved minor comments

reverendbedford reviewed (approved):
Okay, as long as the tests runs.
